weblate/api/views.py

```diff
@@ -1010,6 +1012,41 @@ def file(self, request: Request, **kwargs):
             name=instance.slug,
         )
 
+    @action(detail=True, methods=["get", "post"])
+    def machinery_settings(self, request: Request, **kwargs):
+        """List or create/update machinery configuration for a project."""
+        project = self.get_object()
+        if request.method == "POST":
+            if not request.user.has_perm("project.edit", project):
+                self.permission_denied(request, "Can not edit machinery configuration")
+
+            try:
+                service_name = request.data["service"]
+            except KeyError:
+                return Response(
+                    {"errors": ["Missing service name"]}, status=HTTP_400_BAD_REQUEST
+                )
+
+            service, configuration, errors = validate_service_configuration(
+                service_name, request.data.get("configuration", "{}")
+            )
+            if service is None:
+                return Response(
+                    {"errors": errors},
+                    status=HTTP_400_BAD_REQUEST,
+                )
+
+            if errors:
+                return Response({"errors": errors}, status=HTTP_400_BAD_REQUEST)
+
+            project.machinery_settings[service_name] = configuration
+            project.save(update_fields=["machinery_settings"])
+            return Response(
+                {"message": f"Service installed: {service.name}"}, status=HTTP_200_OK
+            )
+
+        return Response(project.machinery_settings, status=HTTP_200_OK)
```

Comment:
Would it be possible to use a serializer here? That way, the endpoint would be properly documented in OpenAPI.